include/hipSYCL/sycl/libkernel/host/builtins.hpp

```diff
@@ -528,6 +528,31 @@ HIPSYCL_BUILTIN T __hipsycl_clamp(T x, T minval, T maxval) noexcept {
   return std::min(std::max(x, minval), maxval);
 }
 
+template <class T,
+          std::enable_if_t<
+              (std::is_same_v<T, unsigned int> || std::is_same_v<T, int> ||
+               std::is_same_v<T, unsigned short> || std::is_same_v<T, short> ||
+               std::is_same_v<T, unsigned char> ||
+               std::is_same_v<T, signed char> || std::is_same_v<T, char>),
+              int> = 0>
+HIPSYCL_BUILTIN T __hipsycl_clz(T x) noexcept {
+  return __builtin_clz(x);
```

Comment:
It might also be a good idea to use:
```
#if __has_builtin(__builtin_clz)
  (use builtin)
#else
 (some slow fallback)
#endif

I don't know how widespread those builtins are outside of Linux.

Reply:
I've changed it to 

```c++
  #if __has_builtin(__builtin_clz)
    return __builtin_clz(x);
  #else
    // mbit = 1000 0000 ...
    constexpr T mBit = 1 << ((CHAR_BIT*sizeof(T)) - 1);
    T count = 0;
    while (!(x & mBit))
    {
      x = (x << 1);
      count ++;
    }
    return count;
  #endif
```